api/database/dbConnection.ts

```diff
@@ -64,4 +64,50 @@ const setupDatabaseConnection = (
   return db;
 };
 
-export default setupDatabaseConnection;
+export const createDatabaseIfNotExists = async (
+  dbName: string,
+  defaultDb: string | undefined,
+  host: string | undefined,
+  user: string | undefined,
+  password: string | undefined,
+  port: string,
+  ssl: string | undefined,
+): Promise<boolean> => {
+  if (!dbName) {
+    throw new Error("Database name is required");
+  }
+
+  const client = new Client({
+    user: user,
+    host: host,
+    password: password,
+    port: parseInt(port, 10),
+    ssl: ssl === "true" ? { rejectUnauthorized: false } : false,
+    database: defaultDb,
+  });
+
+  try {
+    await client.connect();
```

Comment:
When does this run?  Looks like it runs on module import time, which is not necessarily the same as server startup.

I am not very familiar at this level of Javascript, but typically you want code import to go really fast.  Also, I imagine you couldn't import this file into a test without the import failing?

I would expect this to be called from either
- some sort of explicit  HTTP server-startup function, or
- a separate CLI command that gets called BEFORE the server-startup CLI command.  Like you might call the former only when there are database migrations (you  might call it "migrate")

Reply:
My mistake! I missed the createDatabaseIfNotExists definition and 2 space of indent and was thinking this was at the top-level.